### ISSUE TITLE
dispute-distribution should be a noop for full nodes

### ISSUE THREAD

**Issue by eskimor** [paritytech/polkadot#6462]
We seem to be importing votes in the dispute-coordinator, even on simply full nodes (e.g. collators). This in itself is unnecessary and should likely be removed. One artifact of this is that dispute-distribution learns about those unknown disputes and tries to find the node's vote in order to send it out to validators.

Both subsystems should ignore disputes further up the stack in order to not waste resources. In particular dispute-distribution is effectively doing nothing already, but it does "nothing" in a loop which is wasteful, instead the full loop should be skipped.

In particular we should check whether the node has validator keys in the keystore and only if it has at least one, we enable functionality on active leaves updates. Keys being present is enough to enable functionality. In particular, we should not check whether we are an active validator in the current session. This is because, we could become active in the next session for example and then having recorded dispute votes is beneficial to the network's security.

**Comment by bkchr** [comment]
@skunert wants to clean up subsystems usage in Cumulus any way. We should probably have dispute distribution not even registered on collators/full nodes if possible?

**Comment by eskimor** [comment]
Yes, that would be even better, although not as straight forward. For example relay-chain-selection is sending messages to the dispute-coordinator to find out about disputed. Given that full nodes don't vote on finality this in itself is also pointless. So yes, cleaning this up on a deeper level makes a lot of sense, but might also be a bit involved.

**Comment by eskimor** [comment]
Even better than checking for keys, would be to just tell the subsystem whether the node is meant to be an authority or not, like we do it for [network protocols](https://github.com/paritytech/polkadot/blob/ac0d5ab5448303dc834d2afca6c29c4d93920859/node/service/src/lib.rs#L878).

**Comment by eskimor** [comment]
@BradleyOlson64 also an easier one to tackle. While a larger scale cleanup as suggested by @bkchr is a very good idea, I think an intermediate solution where we just make sure to not do any unnecessary work would still be worthwhile.
